SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -50,6 +56,74 @@ void GetSubobjects(UObject* Object, TArray<UObject*>& InSubobjects)
 	});
 }
 
+Worker_EntityId USpatialPackageMapClient::AllocateEntityIdForActor(AActor* Actor)
+{
+	check(Actor);
+	checkf(NetDriver->IsServer(), TEXT("Tried to allocate an Entity ID on the client, this shouldn't happen."));
+
+	Worker_EntityId EntityId = NetDriver->EntityPool->GetNextEntityId();
+	if (EntityId == SpatialConstants::INVALID_ENTITY_ID)
+	{
+		UE_LOG(LogSpatialOSNetDriver, Error, TEXT("Unable to retrieve an Entity ID for Actor: %s"), *Actor->GetName());
+		return EntityId;
+	}
+
+	NetDriver->GetEntityRegistry()->AddToRegistry(EntityId, Actor);
+
+	// Register Actor with package map since we know what the entity id is.
+	ResolveEntityActor(Actor, EntityId);
```

Comment:
Hmmm. this function doesn't give any indication that it's going to be resolving the Actor after getting an entity Id.

I'm thinking, the code surrounding EntityPool/PackageMap shouldn't have to deal with "getting" an entity id, and only "resolving". What do you think of having a `TryResolveEntityActor` which is called in the ActorChannel, rather than TryAllocateEntityId. This potentially could be a rename of the function and moving ResolveEntityActor in there.

Reply:
Not sure what you mean exactly.
This functionality can't live in the ActorChannel because the channel may not have been created yet, e.g. if we created 2 actors pointing at each other on the same frame, then one channel gets created first and needs to replicate a reference to the second actor which doesn't have a channel yet.
I do agree about the naming, how about `AllocateEntityIdAndResolveActor`?